controller/api_0.0.0/video.js

```diff
@@ -0,0 +1,62 @@
+let router          = require('express').Router();
+let op              = require('../../model/api_operations');
+let getVideoInfo    = require('../function/video_info');
+
+router.post(
+    '/api/0.0.0/submit_video',
+    require('../../middleware/logged_in_only.js'),
+    (req, res) => {
+        // verify all the data come thru
+        // need to have the url, by, and channel
+
+        let url = req.body.url;
+        let who_submit = req.body.who_submit;
+        let which_channel = req.body.which_channel;
+
+        getVideoInfo(url).then(function (result) {
+            // if we are able to get the video's info
+            const ret_value = op.submit_video(result.id, which_channel, result.length, who_submit);
+            if (ret_value[1]) { // if created
```

Comment:
I did a test, ret_value[1] returned undefined

Reply:
Will check when I get home

Reply:
@jiajern `const ret_value = op.submit_video(result.id, which_channel, result.length, who_submit);` wouldn't work. Since `op.submit_video(...)` is doing a asynchronous operation, the result is not immediately available to be assigned to `ret_value`. `submit_video()` returns promise, so you'll have to use `then/catch` or something similar.

Reply:
Ok got it.